unstructured/partition/pdf.py

```diff
@@ -797,91 +789,64 @@ def _add_pytesseract_bboxes_to_elements(
-                lang=ocr_languages,
+        images = []
+        image = PILImage.open(file) if file is not None else PILImage.open(filename)
+        images.append(image)
+
+        for i, image in enumerate(images):
+            page_layout = get_page_layout_from_ocr(
+                image=image,
+                page_number=i + 1,
+                ocr_languages=ocr_languages,
+                ocr_agent=ocr_agent,
             )
-        elements = partition_text(
-            text=text,
-            max_partition=max_partition,
-            min_partition=min_partition,
-            metadata_last_modified=metadata_last_modified,
-            detection_origin="OCR",
-        )
-        width, height = image.size
-        _add_pytesseract_bboxes_to_elements(
-            elements=cast(List[Text], elements),
-            bboxes_string=_bboxes,
-            width=width,
-            height=height,
-        )
-
+            page_layouts.append(page_layout)
     else:
-        elements = []
         page_number = 0
         for image in convert_pdf_to_images(filename, file):
             page_number += 1
-            metadata = ElementMetadata(
-                filename=filename,
+            page_layout = get_page_layout_from_ocr(
+                image=image,
                 page_number=page_number,
-                last_modified=metadata_last_modified,
-                languages=languages,
-            )
-            metadata.detection_origin = "OCR"
-            _text, _bboxes = unstructured_pytesseract.run_and_get_multiple_output(
-                image,
-                extensions=["txt", "box"],
-                lang=ocr_languages,
-            )
-            width, height = image.size
-
-            _elements = partition_text(
-                text=_text,
-                max_partition=max_partition,
-                min_partition=min_partition,
+                ocr_languages=ocr_languages,
+                ocr_agent=ocr_agent,
             )
+            page_layouts.append(page_layout)
 
-            for element in _elements:
-                element.metadata = metadata
+    layout = DocumentLayout.from_pages(page_layouts)
 
-            _add_pytesseract_bboxes_to_elements(
-                elements=cast(List[Text], _elements),
-                bboxes_string=_bboxes,
-                width=width,
-                height=height,
-            )
+    elements = document_to_element_list(
+        layout,
+        sortable=True,
+        include_page_breaks=include_page_breaks,
+        last_modification_date=metadata_last_modified,
+        # NOTE(crag): do not attempt to derive ListItem's from a layout-recognized "List"
+        # block with NLP rules. Otherwise, the assumptions in
+        # unstructured.partition.common::layout_list_to_list_items often result in weird chunking.
+        infer_list_items=False,
+        detection_origin="image" if is_image else "pdf",
+        languages=languages,
+        **kwargs,
+    )
```

Comment:
This seems roundabout to me... we're putting it in `unstructured-inference` format only to immediately convert it back to `unstructured` format?

What's the motivation there?

Reply:
The motivation is to have a pattern similar to the `hi_res` strategy.